### ISSUE TITLE
no-unsupported-features/node-builtins should allow checking if builtin is defined

### ISSUE THREAD

**Issue by gabegorelick** [mysticatea/eslint-plugin-node#189]
`no-unsupported-features/node-builtins` should allow checking if a builtin exists and overwriting a builtin. 

Let's say you're on Node 8 and want to polyfill crypto.scrypt:
```js
const crypto = require('crypto');
if (!crypto.scrypt) { // fails no-unsupported-features/node-builtins
  scrypto.scrypt = userLandScrypt // fails no-unsupported-features/node-builtins
}
```
